openlxp_authentication/models.py

```diff
@@ -0,0 +1,52 @@
+from django.db import models
+from django.db.models.base import Model
+from django.conf import settings
+from social_core.backends.saml import SAMLAuth, SAMLIdentityProvider
+from social_django.strategy import DjangoStrategy
+
+
+class SAMLConf(Model):
+    name = models.CharField(unique=True, max_length=50)
+    entity_id = models.CharField(max_length=300)
+    url = models.URLField()
+    cert = models.TextField()
+
+    def attributes():
+        return dict.fromkeys(["attr_" + attr for attr in settings.USER_ATTRIBUTES], '')
+    attribute_mapping = models.JSONField(default=attributes)
+
+
+class SAMLDBAuth(SAMLAuth):
+    name = 'samldb'
+
+    def get_idp(self, idp_name):
+        conf_name = idp_name.split(
+            '/')[-2] if idp_name.split('/')[-1] == '' else idp_name.split('/')[-1]
+        idp_config = SAMLConf.objects.get(name=conf_name)
+        config_map = {"entity_id": idp_config.entity_id,
+                      "url": idp_config.url, "x509cert": idp_config.cert, **idp_config.attribute_mapping}
+        return SAMLIdentityProvider(conf_name, **config_map)
+
+
+class SAMLDBStrategy(DjangoStrategy):
```

Comment:
As with the SAMLAuth derivation above, we'll. just want to make sure that the `SAMLDB` prefix is revised, accordingly, if we choose to change the name.